### PR TITLE
feat: MaxTokensLimiter extension

### PR DESCRIPTION
**Pull request by reka**
Limit the maximum number of tokens in a GraphQL document.

Implementation
----------------

* `ExecutionContext`: add `parse_options`
* pass `parse_options` to `graphql.language.parser.parse`

Docs
-------

* deployment docs: mention security extensions

Other
----------
max_aliases docstring: fix copy-paste error

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tests/schema/extensions/test_max_tokens.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      }
+    }
+    """
+
+    result = _execute_with_max_tokens(query, 13)
+
+    assert not result.errors
+    assert result.data
+
+
```
Comment (reka) [inline tests/schema/extensions/test_max_tokens.py:58]: Which further test cases do we need? :bulb:

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This PR adds a MaxTokensLimiter extension which limits the number of tokens in a GraphQL document.

## Usage example:

```python
import strawberry
from strawberry.extensions import MaxTokensLimiter

schema = strawberry.Schema(
    Query,
    extensions=[
        MaxTokensLimiter(max_token_count=1000),
    ],
)
```


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to reka for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/types/execution.py [hunk 2]
```diff
@@ -24,6 +24,7 @@ class ExecutionContext:
     schema: Schema
     context: Any = None
     variables: Optional[Dict[str, Any]] = None
+    parse_options: Dict[str, Any] = dataclasses.field(default_factory=lambda: {})
```
Comment (patrick91) [inline strawberry/types/execution.py:27]: let's make a typed dict and reuse it in the function above 😊

**Review by patrick91 — COMMENTED** [review]
Great stuff! I've only added a comment about using typed dicts 😊

File: pyproject.toml [hunk 3]
```diff
@@ -151,7 +151,8 @@ markers = [
     "fastapi",
     "chalice",
     "flask",
-    "starlite"
+    "starlite",
+    "flaky"
```
Comment (reka) [inline pyproject.toml:155]: Is it OK to add the `flaky` marker in that PR?
Or should it go into a separate PR?

Comment (patrick91) [reply pyproject.toml:155]: yes 😊

File: strawberry/types/execution.py [hunk 4]
```diff
@@ -91,5 +90,7 @@ class ExecutionResult:
     extensions: Optional[Dict[str, Any]] = None
 
 
-class ParseOptions(TypedDict):
-    max_tokens: NotRequired[int]
+class ParseOptions(TypedDict, total=False):
+    # Change this to typing.NotRequired
+    # when we don't support versions older than 3.11
```
Comment (patrick91) [inline strawberry/types/execution.py:95]: we can use notrequired from typing extension https://pypi.org/project/typing-extensions/

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -122,6 +122,8 @@ $ poetry run pytest tests -n auto
 $ poetry run mypy
 ```
 
+Some tests are known to be inconsistent. (The fix is in progress.) These tests are marked with the `pytest.mark.flaky` marker.
```
Comment (patrick91) [inline CONTRIBUTING.md:125]: 👍

**Review by patrick91 — APPROVED** [review]
Looks great! Thanks 😊

**Pull request merged**